File: app/src/main/java/com/mdkashif/alarm/custom/AlarmWebViewClient.java
package com.mdkashif.alarm.custom;

import android.graphics.Bitmap;
import android.webkit.WebView;

import com.mdkashif.alarm.activities.BaseActivity;


public class AlarmWebViewClient extends android.webkit.WebViewClient {

    private boolean isErrorRecevied = false;
    boolean isWebviewLoadedCompletely=false;
    BaseActivity baseActivity;

    @Override
    public boolean shouldOverrideUrlLoading(WebView view, String url) {
        view.loadUrl(url);
        isWebviewLoadedCompletely=true;
        return true;
    }

    public AlarmWebViewClient(BaseActivity baseActivity) {
        this.baseActivity = baseActivity;
    }

    @Override
    public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
        isErrorRecevied=true;
    }

    @Override
    public void onPageStarted(WebView view, String url, Bitmap favicon) {
        baseActivity.showLoader();
    }

    public void onPageFinished(final WebView view,  String url) {

        if (isWebviewLoadedCompletely && !isErrorRecevied) {
            baseActivity.hideLoader();
        }
        else {
            baseActivity.hideLoader();
            baseActivity.showToast("Sorry for the Inconvenience, Please try again later");
        }

    }

}

File: app/src/main/java/com/mdkashif/universalarm/custom/AlarmWebViewClient.kt
package com.mdkashif.universalarm.custom

import android.graphics.Bitmap
import android.webkit.WebView

import com.mdkashif.universalarm.activities.BaseActivity

class AlarmWebViewClient(private val baseActivity: BaseActivity) : android.webkit.WebViewClient() {
    private var isErrorReceived = false
    private var isWebViewLoadedCompletely = false

    override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
        view.loadUrl(url)
        isWebViewLoadedCompletely = true
        return true
    }

    override fun onReceivedError(view: WebView, errorCode: Int, description: String, failingUrl: String) {
        isErrorReceived = true
    }

    override fun onPageStarted(view: WebView, url: String, favicon: Bitmap) {
        baseActivity.showLoader()
    }

    override fun onPageFinished(view: WebView, url: String) {
        if (isWebViewLoadedCompletely && !isErrorReceived) {
            baseActivity.hideLoader()
        } else {
            baseActivity.hideLoader()
            baseActivity.showToast("Sorry for the Inconvenience, Please try again later")
        }
    }
}
